### PR TITLE
Adds better NamedTuple error messages

### PR DESCRIPTION
**Pull request by sobolevn**
While working on https://github.com/python/mypy/issues/11047 I've noticed that `mypy` reports the same error for both `collection.namedtuple()` and `typing.NamedTuple()`.

I've tried to fix to report the correct name.

### CODE REVIEW

File: test-data/unit/check-namedtuple.test [hunk 1]
```diff
@@ -35,7 +35,7 @@ X = namedtuple('X', ('x', 'y'))  # type: ignore
 [case testNamedTupleNoUnderscoreFields]
 from collections import namedtuple
 
-X = namedtuple('X', 'x, _y, _z')  # E: namedtuple() field names cannot start with an underscore: _y, _z
+X = namedtuple('X', 'x, _y, _z')  # E: "namedtuple()" field names cannot start with an underscore: _y, _z
```
Comment (JelleZijlstra) [inline test-data/unit/check-namedtuple.test:38]: Why did you add the extra quotes? They feel unnecessary.

Comment (sobolevn) [reply test-data/unit/check-namedtuple.test:38]: Most of the time mypy highlights the most important code part with `""`, this feels more similar to other cases. For example, here's how it renders in my terminal (with / without `""`):
Снимок экрана 2021-09-18 в 1 20 34

In my opinion, the first one looks and reads better.

Comment (JukkaL) [reply test-data/unit/check-namedtuple.test:38]: The double quotes look reasonable to me. Previously we didn't use them probably because the () suffix makes it look like code, it seems okay to add them.

File: mypy/semanal_namedtuple.py [hunk 2]
```diff
@@ -262,16 +262,17 @@ def parse_namedtuple_args(self, call: CallExpr, fullname: str
 
         Return None if the definition didn't typecheck.
         """
+        type_name = 'NamedTuple' if fullname == 'typing.NamedTuple' else 'namedtuple'
         # TODO: Share code with check_argument_count in checkexpr.py?
         args = call.args
         if len(args) < 2:
-            self.fail("Too few arguments for namedtuple()", call)
+            self.fail('Too few arguments for "{0}()"'.format(type_name), call)
```
Comment (JukkaL) [inline mypy/semanal_namedtuple.py:269]: Style nit: We tend to use just `{}` in format strings instead of `{0}` whenever possible (applies to here and elsewhere it this PR). Not a big deal though.

**Review by JukkaL — APPROVED** [review]
Looks good, just one style nit (optional).
